dlauthenticator/dlauthenticator.py

```diff
@@ -226,17 +360,44 @@ def parser_arguments():
     '''Test Application
     '''
     import getpass
+    from unittest.mock import MagicMock, PropertyMock
 
     # Parse the command-line arguments.
     args = parser_arguments()
 
-    if args.user is None and args.password is None:
-        data = dict(username=input('Username: '), password=getpass.getpass())
-    else:
-        data = dict(username=args.user, password=args.password)
-
-    rs = DataLabAuthenticator().authenticate(None, data)
+    #if args.user is None and args.password is None:
+    #    data = dict(username=input('Username: '), password=getpass.getpass())
+    #else:
+    #    data = dict(username=args.user, password=args.password)
+
+#    rs = DataLabAuthenticator().authenticate(None, data)
+#    if rs.result() is None:
+#        print(f'Login fails for user: %s' % data['username'])
+#    else:
+#        print(f'Login succeeds for user: %s' % data['username'])
+    # Step 1: Create a mock request
+
+    #"Cookie": "X-DL-AuthToken = isuarezsola.3195.3195.$1$zihxE5de$CHapp./1cp5zTGdwwMDNi."
+    mock_headers = {
+        "Cookie": "X-DL-AuthToken = testgcuser1.4146.4146.$1$cH8mTs3B$m.WkTNYW2q3PXRCMwuIgB/"
+    }
+    # Starting the patch, and getting a reference to the mock object
+    mock_isValidToken = patch('dl.authClient.isValidToken').start()
+    # Setting the return value for the mock object
+    mock_isValidToken.return_value = True
+
+    mock_request = MagicMock()
+
+    type(mock_request).headers = PropertyMock(return_value=mock_headers)
+
+    # Mocking the handler to have the mocked request as its attribute
+    mock_handler = MagicMock()
+    type(mock_handler).request = PropertyMock(return_value=mock_request)
+
+    data = {'username': 'blah'}
+    rs = GCDataLabAuthenticator().authenticate(mock_handler, {'username': data})
     if rs.result() is None:
         print(f'Login fails for user: %s' % data['username'])
     else:
+        print("GCDataLabAuthenticator:", rs.result())
         print(f'Login succeeds for user: %s' % data['username'])
```

Comment:
We now have a lot of nice looking test files in the tests directory and this is covered in the `test_valid_login` file, can we perhaps just remove this block from the `dlauthenticator` file?

Reply:
In that case, I'll take the whole main section out. Do you see any reason to keep the __main__ section? @chaduhduh

Reply:
Nope!